tests/templates/test_embeddings/test_qaoa_emb.py

```diff
@@ -257,12 +241,47 @@ def test_shape(self, n_layers, n_wires, expected_shape):
         assert shape == expected_shape
 
 
-class TestGradients:
-    """Tests that the gradient is computed correctly in all interfaces."""
+def circuit_template(features, weights):
+    qml.templates.QAOAEmbedding(features, weights, range(2))
+    return qml.expval(qml.PauliZ(0))
+
+
+def circuit_decomposed(features, weights):
+    qml.RX(features[0], wires=0)
+    qml.RX(features[1], wires=1)
+    qml.MultiRZ(weights[0, 0], wires=[0, 1])
+    qml.RY(weights[0, 1], wires=0)
+    qml.RY(weights[0, 2], wires=1)
+    qml.RX(features[0], wires=0)
+    qml.RX(features[1], wires=1)
+    return qml.expval(qml.PauliZ(0))
+
```

Comment:
While seeing this nice way of testing the templates, I think about the length of the UCCSD one :-(. Perhaps, once the ``SingleExcitationUnitary`` and ``DoubleExcitationUnitary`` are tested, for the UCCSD we just need to test that the order is correct :thinking:

Reply:
Yes, that's a good idea!